ledger/src/blockstore.rs

```diff
@@ -2042,9 +2069,22 @@ impl Blockstore {
                     .last()
                     .map(|entry| entry.hash)
                     .unwrap_or_else(|| panic!("Rooted slot {slot:?} must have blockhash"));
+                let mut starting_transaction_index = 0;
+                let mut entries = vec![];
```

Comment:
If we end up going with an `Option<Vec<_>>` here, we could avoid having to grow the vector in the loop.
```rust
let mut entries = if populate_entries {
    Some(Vec::with_capacity(slot_entries.len()))
} else {
    None
};
```
I guess even if we don't go with the `Option<_>`, you could just do an empty vector in the `populate_entries == false` case

Reply:
I mean, it should be an empty vector in the `populate_entries == false` case as written, as any/all pushes to `entries` are gated by `populate_entries`. I don't think the Option really gains us much, unless I'm missing something about the compiler in this case...?

Reply:
Per other comment thread, I now see why the `Option` would be redundant. That being said, I think we can still allocate the correct capacity at the start:
```rust
let mut entries = if populate_entries {
    Vec::with_capacity(slot_entries.len())
} else {
    Vec::new()
};
```

Reply:
Ah, great suggestion. Pushed.